app/models/workers_scaling/algorithm.rb

```diff
@@ -1,25 +1,65 @@
+require_relative 'experiment_resources_interface'
+require_relative 'experiment_statistics'
 module WorkersScaling
   ##
   # Class describing interface of Workers scaling algorithm. Creating new algorithm
   # requires providing class with proper methods implemented, which also should
   # inherit Algorithm class.
-  class Algorithm
+  class Algorithm < Scalarm::Database::MongoActiveRecord
+    use_collection 'workers_scaling_algorithms'
+    attr_accessor :experiment
+    attr_accessor :resources_interface
+    attr_accessor :experiment_statistics
 
     NOT_IMPLEMENTED = 'This is an abstract method, which must be implemented by all subclasses'
+    ALGORITHM_INTERVAL = 30.seconds
 
     ##
-    # Arguments:
-    # * experiment - instance of Experiment
-    # * user_id - id of User starting Algorithm
-    # * allowed_infrastructures - list of hashes with infrastructure and maximal Workers amount
-    #     (Detailed description at ExperimentResourcesInterface#initialize)
-    # * planned_finish_time - desired time of end of Experiment (as Time instance)
-    # * params - additional params, currently unused, may be used in subclasses
-    def initialize(experiment, user_id, allowed_infrastructures, planned_finish_time, params = {})
-      @experiment = experiment
+    # Returns name of Algorithm implementation class
+    def self.get_class_name
+      self.name.gsub('::', '__').underscore.to_sym
+    end
+
+    ##
+    # Finds all records of Algorithm implementations
+    # Returns hash containing {experiment_id => next_execution_time} pair for each record
+    def self.get_all_algorithms_times
```

Comment:
Wyjdzie długie, ale po samej sygnaturze nie wiem co to za czas - czy następnego uruchomienia, czy ostatniej aktualizacji.